### PR TITLE
Fixes #25036 - ensure proxy registration happens before puppet

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
failures seem unrelated

File: manifests/register.pp [hunk 1]
```diff
@@ -11,6 +11,12 @@
       url             => $foreman_proxy::real_registered_proxy_url,
     }
 
+    # Ensure the service is started after registering the Foreman proxy
+    # Relationship is duplicated there as defined() is parse-order dependent
+    if defined(Class['puppet::agent::service']) {
+      Foreman_smartproxy[$foreman_proxy::registered_name] -> Class['puppet::agent::service']
```
Comment (ekohl) [inline manifests/register.pp]: If you rewrite this to `-> Class <| title == 'puppet::agent::service' |>` I believe you don't need the `defined` and it's no longer parse-order dependent.

Comment (evgeni) [reply manifests/register.pp]: [you can't collect `Class`es](https://puppet.com/docs/puppet/5.5/lang_collectors.html) :(

Comment (evgeni) [reply manifests/register.pp]: but you can:

```puppet
    Foreman_smartproxy[$foreman_proxy::registered_name] -> Cron <| title == 'puppet' |>
    Foreman_smartproxy[$foreman_proxy::registered_name] -> Service <| title == 'puppet' |>
    Foreman_smartproxy[$foreman_proxy::registered_name] -> Service <| title == 'puppet-run.timer' |>
```

Comment (evgeni) [reply manifests/register.pp]: Seems to have worked in my test setup.

**Comment by mmoll** [comment]
@evgeni pelase rebase

**Comment by mmoll** [comment]
merged, cпасибо @evgeni!

**Pull request merged**